source/language/pulses.rst

```diff
@@ -103,3 +102,62 @@ circuit. Most of the time symbols in the ``defcal`` table will also have
 corresponding ``gate`` definitions. However, if a user provides a ``defcal`` for a symbol
 without a corresponding ``gate``, then we treat such operations like the ``opaque`` gates
 of prior versions of OpenQASM.
+
+Inline calibration blocks
+~~~~~~~~~~~~~~~~~~~~~~~~~
+
+As calibration grammars may require the ability to insert top-level configuration information, perform program setup, or make inline calls
```

Comment:
Proposal: we only allow `const` on the OpenQASM be referenced by `cal`. Letting `cal` access and modify variables on the OpenQASM level can make it very hard for compiler implementation.

Reply:
If a 'cal' block can be defined inside an OpenQASM loop, I wonder if the code inside `cal` needs to access the loop variable; this appears fairly common whenever we write loops.